### PR TITLE
Added Guild Storage Expansion Skill

### PR DESCRIPTION
**Pull request by aleos89**
* **Addressed Issue(s)**: #499, #1303

* **Server Mode**: Pre-renewal and Renewal

* **Description of Pull Request**: 
  * Implements Guild Storage Expansion Skill which requires client 2013-12-23 or newer (enable by default depending on client date in src/config/packets.hpp).
  * Implements guild storage permission which requires client 2014-02-05 or newer.
  * Expanded script command guildopenstorage return values.

### CODE REVIEW

File: src/common/mmo.hpp [hunk 1]
```diff
@@ -701,6 +701,20 @@ struct guild_castle {
 	int temp_guardians_max;
 };
 
+/// Guild Permissions
+enum e_guild_permission {
+	GUILD_PERM_INVITE	= 0x001,
+	GUILD_PERM_EXPEL	= 0x010,
+#if PACKETVER >= 20140205
+	GUILD_PERM_STORAGE	= 0x100,
```
Comment (secretdataz) [inline src/common/mmo.hpp]: Shouldn't these be 0x001, 0x002, 0x004?

Comment (Lemongrass3110) [reply src/common/mmo.hpp]: Gravity... :-) This is how bitmasks work for one of their developers at least...

Comment (thervingi) [reply src/common/mmo.hpp]: Funny

**Comment by Lemongrass3110** [comment]
The actual change is like this:
`if( (i=guild_getposition(sd))<0 || !(g->position[i].mode&GUILD_PERM_INVITE) )`

**Comment by xTakami** [comment]
Thanks! @Lemongrass3110

**Comment by laziem** [comment]
I got this error

error_

```
        CXX     elemental.cpp
        CXX     guild.cpp
guild.cpp: In function ‘int guild_invite(map_session_data*, map_session_data*)’:
guild.cpp:596:29: error: cannot convert ‘map_session_data*’ to ‘guild*’ for argument ‘1’ to ‘int guild_getposition(guild*, map_session_data*)’
  if( (i=guild_getposition(sd))<0 || !(g->position[i].mode&GUILD_PERM_INVITE) )
                             ^
guild.cpp: In function ‘int guild_expulsion(map_session_data*, int, uint32, uint32, const char*)’:
guild.cpp:808:30: error: cannot convert ‘map_session_data*’ to ‘guild*’ for argument ‘1’ to ‘int guild_getposition(guild*, map_session_data*)’
  if( (ps=guild_getposition(sd))<0 || !(g->position[ps].mode&GUILD_PERM_EXPEL) )
                              ^
Makefile:83: recipe for target 'obj/guild.o' failed
```

**Comment by aleos89** [comment]
Same issue posted just above yours.

**Comment by xTakami** [comment]
The current Guild Storage before this change is 600, what will happen to the people who have objects in their Guild Storage?

**Pull request merged**